src/core/thread/mle_router.cpp

```diff
@@ -1108,9 +1108,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kRoute, sizeof(route), route));
     VerifyOrExit(route.IsValid(), error = kThreadError_Parse);
 
-    if ((GetDeviceState() == kDeviceStateChild &&
-         memcmp(&mParent.mMacAddr, &macAddr, sizeof(mParent.mMacAddr)) == 0) ||
-        GetDeviceState() == kDeviceStateRouter || GetDeviceState() == kDeviceStateLeader)
+    if ((mDeviceMode & ModeTlv::kModeFFD) || (memcmp(&mParent.mMacAddr, &macAddr, sizeof(mParent.mMacAddr)) == 0))
```

Comment:
Thanks @jwhui . 
Do you mean only REED need to ProcessRouteTlv in advertisement from its parent?
I am thinking router or leader or REED might also want to process ProcessRouteTlv in advertisement from neighbors they could hear.  I might should have only kept the following condition
`if (mDeviceMode & ModeTlv::kModeFFD)`
Please correct me if I have some misunderstanding.